jvcl/run/JvListComb.pas

```diff
@@ -570,15 +570,16 @@ procedure TJvImageItem.SetText(const Value: string);
       SavedOwner := FOwner;
       try
         FOwner := nil;
-        S.Delete(Index);
         if (SavedOwner.GetOwner is TJvImageListBox) and (TJvImageListBox(SavedOwner.GetOwner).Sorted) then
-          S.AddObject(Value, Self)
+        begin
+          S.Delete(Index);
+          Index := S.AddObject(Value, Self);
+        end
         else
-          S.InsertObject(Index, Value, Self);
+          S[Index]:= value;
```

Comment:
You are missing a space in front of the `:=` operator and the `Value` parameter has to start with a capital V
Further, the `S.Objects[Index]` should also be set to `Self` to be sure that it does what the original line of code did

Reply:
S.Objects does not have to be set I think. The entry whose text is replaced should already have Self in it and TStrings.Put retains it.

Reply:
While I changed the formatting, I don't completely understand SGlienke's comment. Yes, TStrings.Put does a delete internally so that branch does not need one. Does the other branch which still contains the delete method call need it or not?

If I set Sorted to true it crashes, no matter whether the delete is there or not. Commenting that one out just lets it crash differently. If it is there one gets an AV in TJvImageItem.SetIndex at s.IndexOfObject, if one removes the delete one gets an Index or an Argument out of range exception somewhere else (I think it was in RTL).

Reply:
I added a commit with the formatting changes now which also added System.UITypes to the implementation use clause to suppress that well known compiler warning about impossible optimization.

I wait for further guidance now.

Reply:
My point was that the entire change there is nonsense - the benefit comes from not doing the additional IndexOfObject call.